### PR TITLE
create integration_tests.yml

### PR DESCRIPTION
**Pull request by aquilamacedo**
This commit creates a workflow in GitHub Actions for integration tests

### CODE REVIEW

**Review by davidbtadokoro — CHANGES_REQUESTED** [review]
Hi @aquilamacedo, and thank you so much for implementing this important workflow in the CI!

I reviewed the change and left two simple comments. The workflow executed correctly in GitHub Actions, so I think this validates the PR.

Two points that I think are worth addressing:

- In the commit message subject, I think the prefix `.github: workflows: integration_tests:` fits (run `git log .github`, for reference);
- I see that the workflow failed because of a known issue in the `kw device` integration test. So, see if you can address this issue in another commit, or comment out the failing tests in this commit, without forgetting to describe this in the commit message body and creating a issue for it (issue #1015 is related but is a different problem). Either approach is fine by me 😬

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 3
```
Comment (rodrigosiqueira) [inline .github/workflows/integration_tests.yml:16]: I don't remember why do we need `fetch-depth: 3`. Why not 1? Am I missing something?

Comment (rodrigosiqueira) [reply .github/workflows/integration_tests.yml:16]: @aquilamacedo any comment here?

Comment (aquilamacedo) [reply .github/workflows/integration_tests.yml:16]: Upon closer examination of this option, I now think that a fetch depth of 1 would be more appropriate. There is no necessity to delve into the historical data for these particular operations.

Comment (rodrigosiqueira) [reply .github/workflows/integration_tests.yml:16]: @aquilamacedo please update your PR.

**Comment by rodrigosiqueira** [comment]
Hi @aquilamacedo , thanks a lot for this change. I just merged it in the unstable branch.

**Pull request closed without merging**